[Beta report] 2026.5.12-beta.2: Codex runtime still fails loading codex-native-task-runtime

## Summary

Sanitized beta verification report for `2026.5.12-beta.2`, tested specifically because this build was expected to fix the Codex plugin/runtime issue from the previous beta.

Upgrade result: package update succeeded from `2026.5.12-beta.1` to `2026.5.12-beta.2`, gateway restarted on the new version, `@openclaw/codex` updated to `2026.5.12-beta.2`, doctor migrated the default and cron models to `openai/gpt-*`, channels remained connected, and representative cron jobs remained `ok`. However, a fresh direct agent smoke test still did not run on the Codex/OpenAI primary route. The `openai/gpt-5.5` + `codex` runtime failed with `MODULE_NOT_FOUND`, and the run completed through Anthropic fallback.

## Severity-ranked findings

### High: `openai/gpt-*` routed through `codex` still fails before model execution

Post-update model routing looked correct:

```text
agents.defaults.model.primary = openai/gpt-5.5
main.model.primary = openai/gpt-5.5
cron.model.primary = openai/gpt-5.5
mini-cron.model.primary = openai/gpt-5.4-mini
openai/gpt-* model map entries use agentRuntime.id = codex
cron payloads were normalized to openai/gpt-*
```

Installed package snapshot:

```text
@openclaw/codex version: 2026.5.12-beta.2
peerDependency: openclaw >=2026.5.12-beta.2
```

Direct smoke test result was `status: ok`, but final provider/model was Anthropic, not OpenAI/Codex. Sanitized fallback attempt:

```text
provider: openai
model: gpt-5.5
code: MODULE_NOT_FOUND
error: Cannot find module '<global-openclaw>/dist/plugin-sdk/root-alias.cjs/codex-native-task-runtime'
Require stack:
- <state>/npm/node_modules/@openclaw/codex/dist/run-attempt-*.js
```

The relevant import site in the installed Codex plugin is:

```text
<state>/npm/node_modules/@openclaw/codex/dist/run-attempt-*.js:
import { ... } from "openclaw/plugin-sdk/codex-native-task-runtime";
```

The relevant OpenClaw package file layout is:

```text
<global-openclaw>/dist/plugin-sdk/root-alias.cjs
<global-openclaw>/dist/plugin-sdk/codex-native-task-runtime.js
<global-openclaw>/dist/plugin-sdk/codex-native-task-runtime.d.ts
```

The runtime still appears to resolve the subpath as if `root-alias.cjs` were a directory:

```text
<global-openclaw>/dist/plugin-sdk/root-alias.cjs/codex-native-task-runtime
```

Expected: `openai/gpt-5.5` should execute through the `codex` runtime without fallback.

Actual: `openai/gpt-5.5` fails during module load; the agent only succeeds because Anthropic fallback completes the turn.

### Medium: result metadata is confusing when this fallback happens

The direct smoke response showed:

```text
status: ok
text: expected smoke-test text
final provider: anthropic
final model: claude-opus-4-7
fallbackAttempts: [openai/gpt-5.5 MODULE_NOT_FOUND]
```

But the execution trace also reported:

```text
winnerProvider: anthropic
winnerModel: claude-opus-4-7
fallbackUsed: false
attempts: only anthropic success
```

This is confusing for beta validation because the top-level agent metadata clearly shows the primary route failed and a fallback provider won, while `executionTrace.fallbackUsed` says false and does not include the failed OpenAI attempt. This can make an unhealthy Codex primary route look cleaner than it is.

### Medium/Low: updater still emits stale-binary restart warning

The update itself succeeded:

```text
Before: 2026.5.12-beta.1
After: 2026.5.12-beta.2
```

During restart, the updater printed that config was written by `2026.5.12-beta.2` while the command was still running `2026.5.12-beta.1`, then said the gateway already reported the updated version and skipped a redundant restart. The final gateway did report `2026.5.12-beta.2`, so this was not blocking, but it remains noisy in upgrade validation.

### Low / pre-existing or not attributed to this beta

The following were still present but do not appear newly introduced by this beta:

- restrictive `plugins.allow` / bundled discovery compatibility warning
- message-tool allowlist warning for routed channels
- first-time setup policy warning for one channel
- gateway token SecretRef unavailable in the doctor command path
- orphan transcript files
- bootstrap file near configured size limit
- task audit warnings for old lost/delivery-failed tasks
- deprecated `config.loadConfig()` plugin runtime warning despite `plugins doctor` reporting no plugin issues

## Reproduction outline

1. Start from an OAuth/Codex-authenticated OpenClaw host on `2026.5.12-beta.1` where the previous workaround had restored `openai-codex/gpt-*` model refs.
2. Run `openclaw update --channel beta` and land on `2026.5.12-beta.2`.
3. Observe successful host update and `@openclaw/codex@2026.5.12-beta.2` installation.
4. Observe model normalization to `openai/gpt-*` with `agentRuntime.id = codex`.
5. Run a fresh direct agent smoke test:
   `openclaw agent --agent main --session-id <fresh> --message "Reply exactly: ..." --timeout 180 --json`
6. Observe the OpenAI/Codex primary route fail with `MODULE_NOT_FOUND` for `<global-openclaw>/dist/plugin-sdk/root-alias.cjs/codex-native-task-runtime`.
7. Observe Anthropic fallback complete the turn, so the command returns `status: ok` despite the broken primary route.

## Final tested state

- Installed version: `2026.5.12-beta.2`.
- Gateway: reachable and reporting the new version.
- `@openclaw/codex`: `2026.5.12-beta.2`.
- Models/cron payloads: migrated to `openai/gpt-*`.
- Representative cron jobs: `ok` at the status layer.
- Direct Codex/OpenAI smoke: not healthy; primary route fails with `MODULE_NOT_FOUND` and fallback provider completes the run.

## Expected fix direction

Please check the `openclaw/plugin-sdk/codex-native-task-runtime` subpath export / root-alias resolution path used by `@openclaw/codex`. The installed files exist as siblings under `<global-openclaw>/dist/plugin-sdk/`, but resolution is still attempting `<global-openclaw>/dist/plugin-sdk/root-alias.cjs/codex-native-task-runtime`.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -7,6 +7,7 @@ Docs: https://docs.openclaw.ai
 ### Fixes
 
 - Sessions: redact persisted tool result detail metadata before writing transcripts so diagnostic secrets do not survive tool output redaction. (#80444) Thanks @nimbleenigma.
+- Codex runtime: allow the official installed `@openclaw/codex` package to use its private task-runtime SDK helper, fixing `MODULE_NOT_FOUND` during migrated OpenAI/Codex beta runs.
 - Codex migration: make Enter activate the highlighted checkbox row before continuing, so `Skip for now` and bulk-selection rows work even when planned items start preselected.
 - fix: harden safe-bin argument validation [AI]. (#80999) Thanks @pgondhi987.
 - fix: scan plugin runtime entries during install [AI]. (#80998) Thanks @pgondhi987.

--- a/src/plugins/sdk-alias.test.ts
+++ b/src/plugins/sdk-alias.test.ts
@@ -244,6 +244,28 @@ function writePluginEntry(root: string, relativePath: string) {
   return pluginEntry;
 }
 
+function writeInstalledPluginEntry(params: {
+  installRoot: string;
+  packageName: string;
+  entry?: string;
+}) {
+  const entry = params.entry ?? "dist/index.js";
+  const packageRoot = path.join(
+    params.installRoot,
+    "node_modules",
+    ...params.packageName.split("/"),
+  );
+  const pluginEntry = path.join(packageRoot, entry);
+  mkdirSafeDir(path.dirname(pluginEntry));
+  fs.writeFileSync(
+    path.join(packageRoot, "package.json"),
+    JSON.stringify({ name: params.packageName, type: "module" }, null, 2),
+    "utf-8",
+  );
+  fs.writeFileSync(pluginEntry, 'export const plugin = "installed";\n', "utf-8");
+  return { packageRoot, pluginEntry };
+}
+
 function createUserInstalledPluginSdkAliasFixture() {
   const { fixture, sourcePluginEntryPath, sourceRootAlias, sourceChannelRuntimePath } =
     createPluginSdkAliasTargetFixture();
@@ -631,7 +653,7 @@ describe("plugin sdk alias helpers", () => {
     expect(subpaths).toEqual(["core", "qa-channel", "qa-channel-protocol", "qa-lab", "qa-runtime"]);
   });
 
-  it("adds the non-QA private Codex task runtime subpath only for bundled Codex", () => {
+  it("adds the non-QA private Codex task runtime subpath only for trusted Codex plugins", () => {
     const fixture = createPluginSdkAliasFixture({
       packageExports: {
         "./plugin-sdk/core": { default: "./dist/plugin-sdk/core.js" },
@@ -660,6 +682,16 @@ describe("plugin sdk alias helpers", () => {
       fixture.root,
       bundledPluginFile("demo", "src/index.ts"),
     );
+    const { packageRoot: installedCodexRoot, pluginEntry: installedCodexEntry } =
+      writeInstalledPluginEntry({
+        installRoot: path.join(makeTempDir(), ".openclaw", "npm"),
+        packageName: "@openclaw/codex",
+      });
+    const { packageRoot: installedOtherRoot, pluginEntry: installedOtherEntry } =
+      writeInstalledPluginEntry({
+        installRoot: path.join(makeTempDir(), ".openclaw", "npm"),
+        packageName: "@openclaw/demo",
+      });
 
     const codexSubpaths = withEnv({ OPENCLAW_ENABLE_PRIVATE_QA_CLI: undefined }, () =>
       listPluginSdkExportedSubpaths({
@@ -671,9 +703,27 @@ describe("plugin sdk alias helpers", () => {
         modulePath: sourceOtherEntry,
       }),
     );
+    const installedCodexSubpaths = withCwd(installedCodexRoot, () =>
+      withEnv({ OPENCLAW_ENABLE_PRIVATE_QA_CLI: undefined }, () =>
+        listPluginSdkExportedSubpaths({
+          modulePath: installedCodexEntry,
+          argv1: path.join(fixture.root, "openclaw.mjs"),
+        }),
+      ),
+    );
+    const installedOtherSubpaths = withCwd(installedOtherRoot, () =>
+      withEnv({ OPENCLAW_ENABLE_PRIVATE_QA_CLI: undefined }, () =>
+        listPluginSdkExportedSubpaths({
+          modulePath: installedOtherEntry,
+          argv1: path.join(fixture.root, "openclaw.mjs"),
+        }),
+      ),
+    );
 
     expect(codexSubpaths).toEqual(["codex-native-task-runtime", "core"]);
+    expect(installedCodexSubpaths).toEqual(["codex-native-task-runtime", "core"]);
     expect(otherSubpaths).toEqual(["core"]);
+    expect(installedOtherSubpaths).toEqual(["core"]);
   });
 
   it("does not reuse a non-private cached subpath list after private qa gets enabled", () => {
@@ -844,7 +894,7 @@ describe("plugin sdk alias helpers", () => {
     );
   });
 
-  it("aliases non-QA private plugin-sdk subpaths for bundled runtime source loading", () => {
+  it("aliases non-QA private plugin-sdk subpaths for trusted Codex runtime loading", () => {
     const fixture = createPluginSdkAliasFixture({
       packageExports: {
         "./plugin-sdk/core": { default: "./dist/plugin-sdk/core.js" },
@@ -857,8 +907,16 @@ describe("plugin sdk alias helpers", () => {
       "plugin-sdk",
       "codex-native-task-runtime.ts",
     );
+    const distRootAlias = path.join(fixture.root, "dist", "plugin-sdk", "root-alias.cjs");
+    const distCodexNativeTaskRuntimePath = path.join(
+      fixture.root,
+      "dist",
+      "plugin-sdk",
+      "codex-native-task-runtime.js",
+    );
     const sourceQaRuntimePath = path.join(fixture.root, "src", "plugin-sdk", "qa-runtime.ts");
     fs.writeFileSync(sourceRootAlias, "module.exports = {};\n", "utf-8");
+    fs.writeFileSync(distRootAlias, "module.exports = {};\n", "utf-8");
     fs.writeFileSync(
       path.join(fixture.root, "scripts", "lib", "plugin-sdk-private-local-only-subpaths.json"),
       JSON.stringify(["codex-native-task-runtime", "qa-runtime"], null, 2),
@@ -866,6 +924,11 @@ describe("plugin sdk alias helpers", () => {
     );
     fs.writeFileSync(
       sourceCodexNativeTaskRuntimePath,
+      "export const codexNativeTaskRuntime = true;\n",
+      "utf-8",
+    );
+    fs.writeFileSync(
+      distCodexNativeTaskRuntimePath,
       "export const codexNativeTaskRuntime = true;\n",
       "utf-8",
     );
@@ -878,6 +941,16 @@ describe("plugin sdk alias helpers", () => {
       fixture.root,
       bundledPluginFile("demo", "src/index.ts"),
     );
+    const { packageRoot: installedCodexRoot, pluginEntry: installedCodexEntry } =
+      writeInstalledPluginEntry({
+        installRoot: path.join(makeTempDir(), ".openclaw", "npm"),
+        packageName: "@openclaw/codex",
+      });
+    const { packageRoot: installedOtherRoot, pluginEntry: installedOtherEntry } =
+      writeInstalledPluginEntry({
+        installRoot: path.join(makeTempDir(), ".openclaw", "npm"),
+        packageName: "@openclaw/demo",
+      });
 
     const aliases = withEnv(
       { OPENCLAW_ENABLE_PRIVATE_QA_CLI: undefined, NODE_ENV: undefined },
@@ -887,6 +960,26 @@ describe("plugin sdk alias helpers", () => {
       { OPENCLAW_ENABLE_PRIVATE_QA_CLI: undefined, NODE_ENV: undefined },
       () => buildPluginLoaderAliasMap(sourceOtherPluginEntry),
     );
+    const installedAliases = withCwd(installedCodexRoot, () =>
+      withEnv({ OPENCLAW_ENABLE_PRIVATE_QA_CLI: undefined, NODE_ENV: undefined }, () =>
+        buildPluginLoaderAliasMap(
+          installedCodexEntry,
+          path.join(fixture.root, "openclaw.mjs"),
+          undefined,
+          "dist",
+        ),
+      ),
+    );
+    const installedOtherAliases = withCwd(installedOtherRoot, () =>
+      withEnv({ OPENCLAW_ENABLE_PRIVATE_QA_CLI: undefined, NODE_ENV: undefined }, () =>
+        buildPluginLoaderAliasMap(
+          installedOtherEntry,
+          path.join(fixture.root, "openclaw.mjs"),
+          undefined,
+          "dist",
+        ),
+      ),
+    );
 
     expect(fs.realpathSync(aliases["openclaw/plugin-sdk"] ?? "")).toBe(
       fs.realpathSync(sourceRootAlias),
@@ -894,8 +987,12 @@ describe("plugin sdk alias helpers", () => {
     expect(fs.realpathSync(aliases["openclaw/plugin-sdk/codex-native-task-runtime"] ?? "")).toBe(
       fs.realpathSync(sourceCodexNativeTaskRuntimePath),
     );
+    expect(
+      fs.realpathSync(installedAliases["openclaw/plugin-sdk/codex-native-task-runtime"] ?? ""),
+    ).toBe(fs.realpathSync(distCodexNativeTaskRuntimePath));
     expect(aliases["openclaw/plugin-sdk/qa-runtime"]).toBeUndefined();
     expect(otherAliases["openclaw/plugin-sdk/codex-native-task-runtime"]).toBeUndefined();
+    expect(installedOtherAliases["openclaw/plugin-sdk/codex-native-task-runtime"]).toBeUndefined();
   });
 
   it("applies explicit dist resolution to plugin-sdk subpath aliases too", () => {

--- a/src/plugins/sdk-alias.test.ts
+++ b/src/plugins/sdk-alias.test.ts
@@ -659,10 +659,9 @@ describe("plugin sdk alias helpers", () => {
         "./plugin-sdk/core": { default: "./dist/plugin-sdk/core.js" },
       },
     });
-    fs.writeFileSync(
+    fs.rmSync(
       path.join(fixture.root, "scripts", "lib", "plugin-sdk-private-local-only-subpaths.json"),
-      JSON.stringify(["codex-native-task-runtime", "qa-runtime"], null, 2),
-      "utf-8",
+      { force: true },
     );
     fs.writeFileSync(
       path.join(fixture.root, "src", "plugin-sdk", "codex-native-task-runtime.ts"),
@@ -692,6 +691,15 @@ describe("plugin sdk alias helpers", () => {
         installRoot: path.join(makeTempDir(), ".openclaw", "npm"),
         packageName: "@openclaw/demo",
       });
+    const shadowCodexRoot = path.join(makeTempDir(), ".openclaw", "extensions", "codex-shadow");
+    const shadowCodexEntry = path.join(shadowCodexRoot, "dist", "index.js");
+    mkdirSafeDir(path.dirname(shadowCodexEntry));
+    fs.writeFileSync(
+      path.join(shadowCodexRoot, "package.json"),
+      JSON.stringify({ name: "@openclaw/codex", type: "module" }, null, 2),
+      "utf-8",
+    );
+    fs.writeFileSync(shadowCodexEntry, 'export const plugin = "shadow";\n', "utf-8");
 
     const codexSubpaths = withEnv({ OPENCLAW_ENABLE_PRIVATE_QA_CLI: undefined }, () =>
       listPluginSdkExportedSubpaths({
@@ -719,11 +727,20 @@ describe("plugin sdk alias helpers", () => {
         }),
       ),
     );
+    const shadowCodexSubpaths = withCwd(shadowCodexRoot, () =>
+      withEnv({ OPENCLAW_ENABLE_PRIVATE_QA_CLI: undefined }, () =>
+        listPluginSdkExportedSubpaths({
+          modulePath: shadowCodexEntry,
+          argv1: path.join(fixture.root, "openclaw.mjs"),
+        }),
+      ),
+    );
 
     expect(codexSubpaths).toEqual(["codex-native-task-runtime", "core"]);
     expect(installedCodexSubpaths).toEqual(["codex-native-task-runtime", "core"]);
     expect(otherSubpaths).toEqual(["core"]);
     expect(installedOtherSubpaths).toEqual(["core"]);
+    expect(shadowCodexSubpaths).toEqual(["core"]);
   });
 
   it("does not reuse a non-private cached subpath list after private qa gets enabled", () => {
@@ -917,10 +934,9 @@ describe("plugin sdk alias helpers", () => {
     const sourceQaRuntimePath = path.join(fixture.root, "src", "plugin-sdk", "qa-runtime.ts");
     fs.writeFileSync(sourceRootAlias, "module.exports = {};\n", "utf-8");
     fs.writeFileSync(distRootAlias, "module.exports = {};\n", "utf-8");
-    fs.writeFileSync(
+    fs.rmSync(
       path.join(fixture.root, "scripts", "lib", "plugin-sdk-private-local-only-subpaths.json"),
-      JSON.stringify(["codex-native-task-runtime", "qa-runtime"], null, 2),
-      "utf-8",
+      { force: true },
     );
     fs.writeFileSync(
       sourceCodexNativeTaskRuntimePath,
@@ -951,6 +967,15 @@ describe("plugin sdk alias helpers", () => {
         installRoot: path.join(makeTempDir(), ".openclaw", "npm"),
         packageName: "@openclaw/demo",
       });
+    const shadowCodexRoot = path.join(makeTempDir(), ".openclaw", "extensions", "codex-shadow");
+    const shadowCodexEntry = path.join(shadowCodexRoot, "dist", "index.js");
+    mkdirSafeDir(path.dirname(shadowCodexEntry));
+    fs.writeFileSync(
+      path.join(shadowCodexRoot, "package.json"),
+      JSON.stringify({ name: "@openclaw/codex", type: "module" }, null, 2),
+      "utf-8",
+    );
+    fs.writeFileSync(shadowCodexEntry, 'export const plugin = "shadow";\n', "utf-8");
 
     const aliases = withEnv(
       { OPENCLAW_ENABLE_PRIVATE_QA_CLI: undefined, NODE_ENV: undefined },
@@ -964,6 +989,16 @@ describe("plugin sdk alias helpers", () => {
       withEnv({ OPENCLAW_ENABLE_PRIVATE_QA_CLI: undefined, NODE_ENV: undefined }, () =>
         buildPluginLoaderAliasMap(
           installedCodexEntry,
+          path.join(fixture.root, "openclaw.mjs"),
+          undefined,
+          "dist",
+        ),
+      ),
+    );
+    const shadowCodexAliases = withCwd(shadowCodexRoot, () =>
+      withEnv({ OPENCLAW_ENABLE_PRIVATE_QA_CLI: undefined, NODE_ENV: undefined }, () =>
+        buildPluginLoaderAliasMap(
+          shadowCodexEntry,
           path.join(fixture.root, "openclaw.mjs"),
           undefined,
           "dist",
@@ -993,6 +1028,7 @@ describe("plugin sdk alias helpers", () => {
     expect(aliases["openclaw/plugin-sdk/qa-runtime"]).toBeUndefined();
     expect(otherAliases["openclaw/plugin-sdk/codex-native-task-runtime"]).toBeUndefined();
     expect(installedOtherAliases["openclaw/plugin-sdk/codex-native-task-runtime"]).toBeUndefined();
+    expect(shadowCodexAliases["openclaw/plugin-sdk/codex-native-task-runtime"]).toBeUndefined();
   });
 
   it("applies explicit dist resolution to plugin-sdk subpath aliases too", () => {

--- a/src/plugins/sdk-alias.ts
+++ b/src/plugins/sdk-alias.ts
@@ -310,10 +310,14 @@ function readPrivateLocalOnlyPluginSdkSubpaths(packageRoot: string): string[] {
   const parsed = tryReadJsonSync(
     path.join(packageRoot, "scripts", "lib", "plugin-sdk-private-local-only-subpaths.json"),
   );
-  if (!Array.isArray(parsed)) {
-    return [];
-  }
-  return parsed.filter((subpath): subpath is string => isSafePluginSdkSubpathSegment(subpath));
+  return [
+    ...new Set([
+      CODEX_NATIVE_TASK_RUNTIME_PLUGIN_SDK_SUBPATH,
+      ...(Array.isArray(parsed)
+        ? parsed.filter((subpath): subpath is string => isSafePluginSdkSubpathSegment(subpath))
+        : []),
+    ]),
+  ];
 }
 
 function readBundledPluginPackageName(packageJsonPath: string): string | null {

--- a/src/plugins/sdk-alias.ts
+++ b/src/plugins/sdk-alias.ts
@@ -265,6 +265,7 @@ const cachedPluginSdkScopedAliasMaps = new PluginLruCache<Record<string, string>
   MAX_PLUGIN_LOADER_ALIAS_CACHE_ENTRIES,
 );
 const PLUGIN_SDK_PACKAGE_NAMES = ["openclaw/plugin-sdk", "@openclaw/plugin-sdk"] as const;
+const OFFICIAL_CODEX_PLUGIN_PACKAGE_NAME = "@openclaw/codex";
 const CODEX_NATIVE_TASK_RUNTIME_PLUGIN_SDK_SUBPATH = "codex-native-task-runtime";
 const PLUGIN_SDK_SOURCE_CANDIDATE_EXTENSIONS = [
   ".ts",
@@ -467,6 +468,40 @@ function isBundledCodexPluginModulePath(params: { packageRoot: string; modulePat
   );
 }
 
+function isOfficialInstalledCodexPluginPackageRoot(packageRoot: string) {
+  const segments = path.resolve(packageRoot).split(path.sep).filter(Boolean);
+  const last = segments.at(-1);
+  const scope = segments.at(-2);
+  const nodeModules = segments.at(-3);
+  return last === "codex" && scope === "@openclaw" && nodeModules === "node_modules";
+}
+
+function isOfficialInstalledCodexPluginModulePath(params: { modulePath: string }) {
+  let cursor = path.dirname(path.resolve(params.modulePath));
+  for (let depth = 0; depth < 12; depth += 1) {
+    const packageJson = tryReadJsonSync<{ name?: unknown }>(path.join(cursor, "package.json"));
+    if (packageJson) {
+      return (
+        packageJson.name === OFFICIAL_CODEX_PLUGIN_PACKAGE_NAME &&
+        isOfficialInstalledCodexPluginPackageRoot(cursor)
+      );
+    }
+    const parent = path.dirname(cursor);
+    if (parent === cursor) {
+      break;
+    }
+    cursor = parent;
+  }
+  return false;
+}
+
+function isTrustedCodexPluginModulePath(params: { packageRoot: string; modulePath: string }) {
+  return (
+    isBundledCodexPluginModulePath(params) ||
+    isOfficialInstalledCodexPluginModulePath({ modulePath: params.modulePath })
+  );
+}
+
 function shouldIncludePrivateLocalOnlyPluginSdkSubpath(params: {
   packageRoot: string;
   modulePath: string;
@@ -475,7 +510,7 @@ function shouldIncludePrivateLocalOnlyPluginSdkSubpath(params: {
   return (
     shouldIncludePrivateLocalOnlyPluginSdkSubpaths() ||
     (params.subpath === CODEX_NATIVE_TASK_RUNTIME_PLUGIN_SDK_SUBPATH &&
-      isBundledCodexPluginModulePath({
+      isTrustedCodexPluginModulePath({
         packageRoot: params.packageRoot,
         modulePath: params.modulePath,
       }))
@@ -535,7 +570,7 @@ export function listPluginSdkExportedSubpaths(
   if (!packageRoot) {
     return [];
   }
-  const includeCodexPrivateRuntime = isBundledCodexPluginModulePath({ packageRoot, modulePath });
+  const includeCodexPrivateRuntime = isTrustedCodexPluginModulePath({ packageRoot, modulePath });
   const cacheKey = `${packageRoot}::privateQa=${shouldIncludePrivateLocalOnlyPluginSdkSubpaths() ? "1" : "0"}::codexPrivate=${includeCodexPrivateRuntime ? "1" : "0"}`;
   const cached = cachedPluginSdkExportedSubpaths.get(cacheKey);
   if (cached) {
@@ -573,7 +608,7 @@ export function resolvePluginSdkScopedAliasMap(
     isProduction: process.env.NODE_ENV === "production",
     pluginSdkResolution: params.pluginSdkResolution,
   });
-  const includeCodexPrivateRuntime = isBundledCodexPluginModulePath({ packageRoot, modulePath });
+  const includeCodexPrivateRuntime = isTrustedCodexPluginModulePath({ packageRoot, modulePath });
   const cacheKey = `${packageRoot}::${orderedKinds.join(",")}::privateQa=${shouldIncludePrivateLocalOnlyPluginSdkSubpaths() ? "1" : "0"}::codexPrivate=${includeCodexPrivateRuntime ? "1" : "0"}`;
   const cached = cachedPluginSdkScopedAliasMaps.get(cacheKey);
   if (cached) {

